server/tserver/src/main/java/org/apache/accumulo/tserver/logger/LogFileKey.java

```diff
@@ -189,4 +195,90 @@ public String toString() {
     }
     throw new RuntimeException("Unknown type of entry: " + event);
   }
+
+  /**
+   * Converts LogFileKey to Key. Creates a Key containing all of the LogFileKey fields. The fields
+   * are stored so the Key sorts maintaining the legacy sort order. The row of the Key is composed
+   * of 3 fields: EventNum + tabletID + seq (separated by underscore). The EventNum is the integer
+   * returned by eventType(). The column family is always the event. The column qualifier is
+   * dependent of the type of event and could be empty.
+   *
+   * <pre>
+   *     Key Schema:
+   *     Row = EventNum_tabletID_seq
+   *     Family = event
+   *     Qualifier = tserverSession OR filename OR KeyExtent
+   * </pre>
+   */
+  public Key toKey() throws IOException {
+    String row = "";
+    int eventNum = eventType(event);
+    String family = event.name();
+    String qual = "";
+    switch (event) {
+      case OPEN:
+        row = formatRow(eventNum, 0, 0);
+        qual = tserverSession;
+        break;
+      case COMPACTION_START:
+        row = formatRow(eventNum, tabletId, seq);
+        if (filename != null)
+          qual = filename;
+        break;
+      case MUTATION:
+      case MANY_MUTATIONS:
+      case COMPACTION_FINISH:
+        row = formatRow(eventNum, tabletId, seq);
+        break;
+      case DEFINE_TABLET:
+        row = formatRow(eventNum, tabletId, seq);
+        // Base64 encode KeyExtent
+        DataOutputBuffer buffer = new DataOutputBuffer();
+        tablet.writeTo(buffer);
+        qual = Base64.getEncoder().encodeToString(copyOf(buffer.getData(), buffer.getLength()));
+        buffer.close();
+        break;
+    }
+    return new Key(new Text(row), new Text(family), new Text(qual));
+  }
+
+  // format row = 1_000001_0000000001
+  private String formatRow(int eventNum, int tabletId, long seq) {
+    return String.format("%d_%06d_%010d", eventNum, tabletId, seq);
+  }
```

Comment:
I had commented above about `FastFormat` before seeing this comment. Something like this is probably better than FastFormat since I am making 2 zero padded strings. I also just arbitrarily chose the lengths of tabletId and sequence but this will allow use of all the digits. I will give this a try, thanks.

Reply:
If we are using a byte for row now, I wonder if `new Text(row)` is the best way to create the row for the `Key`. That constructor for Text will copy the bytes.

Reply:
If possible would be better to avoid Text and use byte[].  Could use Key.builder() if there is a need to mix types like byte[] and String when creating a Key.

Reply:
I am working on this change first and once its working, I will get to your other feedback.

There is a `-1` set for the tabletId to check for a bug but it converts to `%ff;%ff;%ff;%ff`, screwing up the sorting. Maybe we don't need the `-1` since its being changed? 
https://github.com/apache/accumulo/blob/7457fe5e0f753f6c9a1289963075c7caac89b45c/server/tserver/src/main/java/org/apache/accumulo/tserver/log/SortedLogRecovery.java#L93-L96

Reply:
FYI it seems to work if I change this value to 0. I am not sure if we use 0 elsewhere though to indicate an initialized state or something but if not, maybe 0 will work to check for this bug.

Reply:
I am not sure what is going on with that -1.  If you want to make negatives sort correctly can use the following tricks. Just do that xor on encode and decode.

https://github.com/apache/accumulo/blob/58a559fa6ac7c5137b5cd1418b0de457f7cad2a9/core/src/main/java/org/apache/accumulo/core/client/lexicoder/LongLexicoder.java#L30

https://github.com/apache/accumulo/blob/58a559fa6ac7c5137b5cd1418b0de457f7cad2a9/core/src/main/java/org/apache/accumulo/core/client/lexicoder/IntegerLexicoder.java#L33

Reply:
It worked, thanks. Cool trick! I implemented your algorithm in 7b12e0e